samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt

```diff
@@ -0,0 +1,246 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.recipes.kt
+
+import java.io.File
+import java.util.logging.Handler
+import java.util.logging.Level
+import java.util.logging.LogRecord
+import java.util.logging.Logger
+import javax.crypto.SecretKey
+import javax.net.ssl.SSLSession
+import javax.net.ssl.SSLSocket
+import okhttp3.Call
+import okhttp3.Connection
+import okhttp3.ConnectionSpec
+import okhttp3.ConnectionSpec.Builder
+import okhttp3.EventListener
+import okhttp3.Handshake
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.TlsVersion.TLS_1_2
+import okhttp3.brotli.BrotliInterceptor
+import okhttp3.dnsoverhttps.DnsOverHttps
+import okio.ByteString.Companion.toByteString
+
+/**
+ * Logs SSL keys to /tmp/key.log, allowing Wireshark to decode traffic and be examined with http2
+ * filter.
+ */
+class WireSharkKeyLoggerListener : EventListener() {
+  var random: String? = null
+  lateinit var currentThread: Thread
+
+  val loggerHandler = object : Handler() {
+    override fun publish(record: LogRecord) {
+      // Try to avoid multi threading issues with concurrent requests
+      if (Thread.currentThread() != currentThread)
+        return
+
+      // https://timothybasanov.com/2016/05/26/java-pre-master-secret.html
+      // https://security.stackexchange.com/questions/35639/decrypting-tls-in-wireshark-when-using-dhe-rsa-ciphersuites
+      // https://stackoverflow.com/questions/36240279/how-do-i-extract-the-pre-master-secret-using-an-openssl-based-client
+
+      // TLSv1.2 Events
+      // Produced ClientHello handshake message
+      // Consuming ServerHello handshake message
+      // Consuming server Certificate handshake message
+      // Consuming server CertificateStatus handshake message
+      // Found trusted certificate
+      // Consuming ECDH ServerKeyExchange handshake message
+      // Consuming ServerHelloDone handshake message
+      // Produced ECDHE ClientKeyExchange handshake message
+      // Produced client Finished handshake message
+      // Consuming server Finished handshake message
+      // Produced ClientHello handshake message
+      //
+      // Raw write
+      // Raw read
+      // Plaintext before ENCRYPTION
+      // Plaintext after DECRYPTION
+      val message = record.message
+
+      val parameters = record.parameters
+
+      if (parameters != null && !message.startsWith("Raw") && !message.startsWith("Plaintext")) {
+//        p(record.message)
+//        p(record.parameters[0])
+        val param = parameters[0] as String
```

Comment:
param –> parameter

(in my goal to reduce the total number of names wherever possible, param/parameters is nails on chalkboard)